src/wasm/wasm-s-parser.cpp

```diff
@@ -1993,10 +2023,11 @@ SExpressionWasmBuilder::makeTryOrCatchBody(Element& s, Type type, bool isTry) {
   if (isTry && !elementStartsWith(s, "do")) {
     throw ParseException("invalid try do clause", s.line, s.col);
   }
-  if (!isTry && !elementStartsWith(s, "catch")) {
+  if (!isTry &&
+      !(elementStartsWith(s, "catch") || elementStartsWith(s, "catch_all"))) {
```

Comment:
nit: I think this would be at least as easy to read if you distributed the `!`.

Reply:
Done